grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -1141,4 +1142,46 @@ class ProductService {
         }
         return results
     }
+
+    def searchProductDtos(String[] terms) {
+        def query = """
+select p.id, p.name, p.product_code as productCode, p.cold_chain as coldChain, p.controlled_substance as controlledSubstance, p.hazardous_material as hazardousMaterial, p.reconditioned,
+ (select max(pc.color) from product_catalog_item pci left outer join product_catalog pc on pci.product_catalog_id = pc.id where pci.product_id = p.id group by pci.product_id) as productColor
+ from product p
+ left outer join product_supplier productsup1_ on p.id=productsup1_.product_id
+ left outer join party manufactur2_ on productsup1_.manufacturer_id=manufactur2_.id and (${terms.collect { "lower(manufactur2_.name) like '%${it}%'" }.join(" or ")})
+ left outer join party supplier_a3_ on productsup1_.supplier_id=supplier_a3_.id and (${terms.collect { "lower(supplier_a3_.name) like '%${it}%'" }.join(" or ")})
+ left outer join inventory_item inventoryi4_ on p.id=inventoryi4_.product_id and (${terms.collect { "lower(inventoryi4_.lot_number) like '%${it}%'" }.join(" or ")})
+ 
+ where p.active=1 and
+ (${terms.collect { """lower(p.name) like '%${it}%' or lower(p.product_code) like '%${it}%' or lower(p.description) like '%${it}%'
```

Comment:
We only use the wildcard on both sides for name and description.

Reply:
Also we're using case-insensitive like (ilike) which is implemented in MySQL by wrapping the column with LOWER() function.
```
LOWER(this_.name) LIKE '%mask%'
```